### PR TITLE
Added the ability for the user to adjust the tolerance and angular tolerance of VRML tessellation

### PR DESCRIPTION
**Pull request by jmwright**
With the KiCAD 3D model generator the generated VRML files were 6MB, which was too large compared to the ~200KB original files. I added the ability for the user to adjust the `tolerance` and `angularTolerance` so that I could decrease the file size. I don't think I'll be able to keep the same quality at ~200MB, but I can get vast improvements in file size with these changes.

### CODE REVIEW

**Comment by jmwright** [comment]
I think this is ready for review now. @adam-urbanczyk and @lorenzncode I requested reviews from you two, but if either of you don't have time to review let me know and I'll request from someone else.

**Review by adam-urbanczyk — APPROVED** [review]

File: .gitignore [hunk 1]
```diff
@@ -19,3 +19,9 @@ simple.step
 simple.stp
 simple.xml
 test.brep
+nested.caf
```
Comment (lorenzncode) [inline .gitignore:22]: I think this is fine.  Just to add I was thinking any new tests for exporters should be implemented with pytest to make use of tmp_path_factor.  Then we wouldn't have this problem.

For example:
https://github.com/lorenzncode/cadquery/blob/e53bdc0951e67da23cccf31f5b334022b23af654/tests/test_exporters.py#L220-L227

**Review by lorenzncode — CHANGES_REQUESTED** [review]
Let's expose tolerance and angular tolerance to `Assembly.save` as well.
https://github.com/CadQuery/cadquery/blob/2eeabc8bd911fe5e1ea5ec80743fb1c712bb8d4e/cadquery/assembly.py#L445
`            exportVRML(self, path, tolerance, angularTolerance)`

File: .gitignore (hunk 1 above)
Comment (jmwright) [reply .gitignore:22]: Should we create an issue for converting the existing tests as well?

Comment (jmwright) [reply .gitignore:22]: I'm not sure I understand what you mean about exposing the tolerances to `Assembly.save`. They [seem to be already exposed](https://github.com/CadQuery/cadquery/blob/9e9b452e66f7925f7c15689f3e07da47b13f2454/cadquery/assembly.py#L422), unless you meant somewhere above or below that level in code.

Comment (lorenzncode) [reply .gitignore:22]: > Should we create an issue for converting the existing tests as well?

Yes, makes sense.  I can follow up and file a separate issue.


> I'm not sure I understand what you mean about exposing the tolerances to Assembly.save. 

You are correct.  The tolerances are already exposed.  I meant to say that the tolerances are not currently passed on to `exportVRML`.

https://github.com/CadQuery/cadquery/blob/a1f46463ce3f5eab25ff2c777918705ff03912cb/cadquery/assembly.py#L444-L445

The request is to pass `tolerance`, and `angularTolerance` to the `exportVRML` call here.  Sorry that wasn't clear.

Comment (jmwright) [reply .gitignore:22]: @lorenzncode I've made that change now. The only oddity I see with exposing that there is that the default `tolerance` and `angularTolerance` values in the `Assembly.save` method signature are set up for GLTF, which are 0.1 and 0.1. This is different than the defaults for `exportVRML`, which are 1e-3 and 0.1 respectively. I don't think it's a huge problem, and I think it's better to have those settings exposed, but in some situations the user might get lower resolution than expected, and than was intended by the original dev who implemented `Assembly.save`, which I believe is @adam-urbanczyk .

Comment (lorenzncode) [reply .gitignore:22]: Given `exportVRML` accepts tolerance settings now, and `Assembly.save` provides these parameters I think the user would expect the tolerance settings to be considered for all formats that accept them.

> This is different than the defaults for exportVRML, which are 1e-3 and 0.1 respectively.

I see, agree better to pass the tolerance settings.  Perhaps GLTF and VRML tolerances could be changed to be consistent if necessary.

**Review by lorenzncode — APPROVED** [review]

**Comment by jmwright** [comment]
@lorenzncode Fixed CI, so I cna merge this now. Thanks for the reviews.

**Pull request merged**